python/lsst/ctrl/bps/htcondor/lssthtc.py

```diff
@@ -421,6 +425,112 @@ def htc_write_condor_file(filename, job_name, job, job_attrs):
+        starting from version 8.9.8.
+
+        Parameters
+        ----------
+        **kwargs
+            Any keyword arguments that Schedd.history(), Schedd.query(), and
+            Schedd.xquery() accepts.
+
+        Returns
+        -------
+        kwargs : `dict` [`str`, Any]
+             Keywords arguments that are guaranteed to work with the Python
+             HTCondor API.
+        """
+        translation_table = {
+            "constraint": "requirements",
+            "projection": "attr_list",
+        }
+        for new, old in translation_table.items():
+            try:
+                kwargs[old] = kwargs.pop(new)
+            except KeyError:
+                pass
+        return kwargs
+
+else:
+
+    def htc_tune_schedd_args(**kwargs):
+        """Ensure that arguments for Schedd are version appropriate.
+
+        This is the fallback function if no version specific alteration are
+        necessary. Effectively, a no-op.
+
+        Parameters
+        ----------
+        **kwargs
+            Any keyword arguments that Schedd.history(), Schedd.query(), and
+            Schedd.xquery() accepts.
+
+        Returns
+        -------
+        kwargs : `dict` [`str`, Any]
+             Keywords arguments that were passed to the function.
+        """
+        return kwargs
+
+
+def htc_query_history(schedds, **kwargs):
+    """Fetch history records from the condor_schedd daemon.
+
+    Parameters
+    ----------
+    schedds : `htcondor.Schedd`
+        HTCondor schedulers which to query for job information.
+    **kwargs
+        Any keyword arguments that Schedd.history() accepts.
+
+    Yields
+    ------
+    schedd_name : `str`
+        Name of the HTCondor scheduler managing the job queue.
+    job_ad : `dict` [`str`, Any`]
+        HTCondor classad describing a job.
+    """
+    # If not set, provide defaults for positional arguments.
+    kwargs.setdefault("constraint", None)
+    kwargs.setdefault("projection", [])
+    kwargs = htc_tune_schedd_args(**kwargs)
+    for schedd_name, schedd in schedds.items():
+        for job_ad in schedd.history(**kwargs):
+            yield schedd_name, job_ad
+
+
+def htc_query_present(schedds, **kwargs):
+    """Query the condor_schedd daemon for job ads.
+
+    Parameters
+    ----------
+    schedds : `htcondor.Schedd`
+        HTCondor schedulers which to query for job information.
+    **kwargs
+        Any keyword arguments that Schedd.xquery() accepts.
+
+    Yields
+    ------
+    schedd_name : `str`
+        Name of the HTCondor scheduler managing the job queue.
+    job_ad : `dict` [`str`, Any`]
+        HTCondor classad describing a job.
+    """
+    kwargs = htc_tune_schedd_args(**kwargs)
+    queries = [schedd.xquery(**kwargs) for schedd in schedds.values()]
+    for query in htcondor.poll(queries):
+        schedd_name = query.tag()
+        for job_ad in query.nextAdsNonBlocking():
+            yield schedd_name, job_ad
+
+
```

Comment:
As far as I can tell `htcondor.version()` (and in turn `htc_version()`) returns the client side version of HTCondor API in use. So if you don't object, I can rewrite`htc_version()` to just return HTC_VER instead of parsing the output of the `htcondor.version()` or get rid of it entirely (it is used once).